packages/picasso/src/CarouselGradient/CarouselGradient.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useMemo } from 'react'
+import cx from 'classnames'
+import { makeStyles, Theme } from '@material-ui/core/styles'
+
+import gradientWidth from './utils/gradientWidth'
+import styles from './styles'
+
+const useStyles = makeStyles<Theme>(styles, { name: 'CarouselGradient' })
+
+type Props = {
+  slidesToShow: number
+  isLastPage: boolean
+}
+
+const CarouselGradient = ({ isLastPage, slidesToShow }: Props) => {
+  const classes = useStyles()
+
+  const hasGradient = !Number.isInteger(slidesToShow)
+
+  const showNextGradient = hasGradient && !isLastPage
+  const showPrevGradient = hasGradient && isLastPage
+
+  const gradientStyle = useMemo(() => {
+    if (!hasGradient) {
+      return {}
+    }
+
+    return {
+      width: `${gradientWidth(slidesToShow)}%`,
+    }
+  }, [slidesToShow, hasGradient])
+
+  if (!hasGradient) {
+    return null
+  }
```

Comment:
```suggestion
  const hasGradient = !Number.isInteger(slidesToShow)
  if (!hasGradient) {
    return null
  }

  const gradientWidth = `${gradientWidth(numSlidesToShow)}%`
  const showNextGradient = !isLastPage
  const showPrevGradient = isLastPage

```

wdyt?
with memoized component, we can avoid using `useMemo` for `gradientWidth`

Reply:
I don't think so, `gradientWidth` is recalculated only when `slidesToShow` changes, which in the real app will never be. Even if we memoize the component, it will still rerender when `isLastPage` changes.